scripts/docker-integration-tests/prometheus/test.sh

```diff
@@ -391,6 +391,47 @@ function test_series {
     '[[ $(curl -s "0.0.0.0:7201/api/v1/series?match[]=prometheus_remote_storage_samples_total&start=-292273086-05-16T16:47:06Z&end=292277025-08-18T07:12:54.999999999Z" | jq -r ".data | length") -eq 1 ]]'
 }
 
+function test_labels {
+  TAG_NAME_0="name_0" TAG_VALUE_0="value_0_1" \
+    TAG_NAME_1="name_1" TAG_VALUE_1="value_1_1" \
+    TAG_NAME_2="name_2" TAG_VALUE_2="value_2_1" \
+    prometheus_remote_write \
+    label_metric now 42.42 \
+    true "Expected request to succeed" \
+    200 "Expected request to return status code 200"
+
+  TAG_NAME_0="name_0" TAG_VALUE_0="value_0_2" \
+    TAG_NAME_1="name_1" TAG_VALUE_1="value_1_2" \
+    prometheus_remote_write \
+    label_metric_2 now 42.42 \
+    true "Expected request to succeed" \
+    200 "Expected request to return status code 200"
+
+  # Test label search with match
+  ATTEMPTS=5 TIMEOUT=2 MAX_TIMEOUT=4 retry_with_backoff  \
+    '[[ $(curl -s "0.0.0.0:7201/api/v1/labels?start=0&end=9999999999999.99999" | jq -r ".data | length") -gt 3 ]]'
```

Comment:
Hm this could start failing if any other global labels are used in metrics sent to the coordinator I believe if `match[]` is not specified?

Maybe it would be better to see if the values are contained by the result or not.

I think you can use a reduce and filter statement to get a unique count of the terms you expect and compare against that?

Might end up with a long line obviously but I think that this is probably better than flakey tests. Here's an example:
```shell
$ echo '["foo","bar","baz"]' | jq -Mc 'reduce (.[] | select(. == "foo" or . == "bar")) as $item ({}; .[$item] += 1)'
{"foo":1,"bar":1}
```

Reply:
ah that was the intention of >= 3. Still being explicit would be good so yeah will use that.